### DISCUSSION TITLE
What should be the aim of a speed limit overlay?

### DISCUSSION THREAD

**Discussion by arrival-spring** [streetcomplete/StreetComplete#5077] [General]
Should it ask as little as possible of the user, or should it ask as much as is needed in order for a data consumer to be able to determine the speed limit? See [Default Speed Limits](https://wiki.openstreetmap.org/wiki/Default_speed_limits)

Really I mean for when there is no sign.

At the one end we could just add `maxspeed:signed=no` and leave it, but that might well not be enough. At the other end we ask urban/rural and then may also need to ask single/dual carriageway. I would only ask about things that SC can't otherwise add. So things like lanes, surface, sidewalk etc. wouldn't be asked at this point (even though they are needed in some countries), but e.g. dual carriageway can't otherwise be determined for certain.

I am coming from a GB perspective, where we have `GB:nsl_restricted`, `GB:nsl_single`, `GB:nsl_dual` and `GB:motorway` which have a one-to-one mapping to the possible implicit speed limits, so my expectation would be towards the 'ask everything' end of the spectrum. However, very few other countries are in the same situation.

**Comment by matkoniecz** [comment]
> However, very few other countries are in the same situation.

What happens in other countries? As far as I know splits may be different but there are typical various classes of default speed limits.

Overall I would ask similar data to existing speed limit quest (though not sure how to deal with USA)

**Comment by arrival-spring** [comment]
What I meant was about how the tagging relates to the default speed limits.

e.g. in Poland the commonly used tags are: `PL:urban`, `PL:rural`, `PL:living_street`, `PL:trunk`, `PL:motorway` and various `PL:zone...`

But the road types are `living street`, `urban`, `rural`, `rural dual carriageway with 2 or more lanes in each direction`, `motorroad with single carriageway`, `motorroad with dual carriageway`, `motorway`

And this isn't quite one-to-one.

Some places have e.g. both `urban school zone` and `rural school zone`, so in such places, if the user selects 'school zone sign' should they then also be asked if it is an urban/rural road?

For USA, I asked on Slack a few weeks back whether they wanted `maxspeed:signed` or `maxspeed:type` and unanimous response was `maxspeed:signed`.

**Comment by arrival-spring** [comment]
Ping @westnordost, I would appreciate your thoughts when you have a chance.

Would you prefer to always ask urban/rural where needed? And to also ask for things like dual carriageway where needed? Or to propose and use `maxspeed:signed` even where there is not enough other information to determine the speed limit with certainty?

**Comment by westnordost** [comment]
Well, certainly adding as much as is needed for data consumers to be able to determine the speed limit seems most useful at first glance.
 
However, for the app users it is of course less convenient as they are asked about all kind of seemingly unrelated questions in case there are no signs. Even whether it is urban or rural is silly to ask in the city but sometimes somewhat difficult to answer in the periphery. And this of course also has to be displayed somewhere somehow in the UI.

Know, that router libraries already have a way to determine more or less reliably if a road is within a built-up area (GB "restricted area") or not. Both GraphHopper and Valhalla do this by analyzing the density of roads, and probably more. These two, by the way, are the only router library projects that so far implemented or planned to implement using the data from the [Default Speed Limits](https://wiki.openstreetmap.org/wiki/Default_speed_limits) wiki page anyway.

In any case, I think the main issue is the **scope**. You've already noticed how incredibly complex the maxspeed topic is, even without implicit speed limits. And to represent this in a clear UI is already a big enough challenge as it is.

Now, if you look at not just one country, but all, you see that there are many many factors that contribute to which default speed limit is applicable. You already know the wiki page. So, I expect that to properly cover all that is going to be a huge task, actually, probably even an ongoing one due to changes in legislation, corrections of the aforementioned wiki page or new (and uncommon) ways how this or that situation will be tagged.

If you are after my opinion, then I'd say: Keep the scope small. The bigger a project gets, the higher the chance that it never gets finished

**Comment by westnordost** [comment]
Slightly related: https://github.com/westnordost/osm-legal-default-speeds/discussions/15
